zerver/views/auth.py

```diff
@@ -330,6 +330,8 @@ def login_page(request, **kwargs):
     if is_subdomain_root_or_alias(request) and settings.REALMS_HAVE_SUBDOMAINS:
         redirect_url = reverse('zerver.views.registration.find_my_team')
         return HttpResponseRedirect(redirect_url)
+    if request.user.is_authenticated():
```

Comment:
I just kind of appended it. I too think that it should be before the subdomain function.
Anyways waiting for @timabbott to instruct me further.